cache_system/src/loader/test_util.rs

```diff
@@ -0,0 +1,239 @@
+use std::{collections::HashMap, fmt::Debug, hash::Hash, sync::Arc};
+
+use async_trait::async_trait;
+use parking_lot::Mutex;
+use tokio::sync::{Barrier, Notify};
+
+use super::Loader;
+
+#[derive(Debug)]
+enum TestLoaderResponse<V> {
+    Answer { v: V, block: Option<Arc<Barrier>> },
+    Panic,
+}
+
+/// An easy-to-mock [`Loader`].
+#[derive(Debug, Default)]
+pub struct TestLoader<K, V, Extra>
```

Comment:
I wonder if giving these types defaults would allow the rest of the code to avoid having to write as complex types (like `Arc<TestLoader<u8, String, ()>>,`)

```suggestion
pub struct TestLoader<K=u8, V=String, Extra=()>
```

Reply:
done